synapse/storage/databases/main/event_federation.py

```diff
@@ -1232,10 +1232,17 @@ def get_forward_extremeties_for_room_txn(txn: LoggingTransaction) -> List[str]:
             txn.execute(sql, (stream_ordering, room_id))
             return [event_id for event_id, in txn]
 
-        return await self.db_pool.runInteraction(
+        event_ids = await self.db_pool.runInteraction(
             "get_forward_extremeties_for_room", get_forward_extremeties_for_room_txn
         )
 
+        # If we didn't find any IDs, then we must have cleared out the
+        # associated `stream_ordering_to_exterm`.
+        if not event_ids:
+            raise StoreError(400, "stream_ordering too old %s" % (stream_ordering,))
+
+        return event_ids
+
```

Comment:
Is this a new error condition, i.e. was it impossible for the interaction to return a falsey value before?

I think before this could fail, but we'd try to keep it working for the last month's worth of stream orderings?

Reply:
It's not, and note that the one caller treats empty list and `StoreError` the same.

This is and was hit when we asked for the extremities at a recent ish stream ordering where we had purged all entries in the room before that stream ordering (we'd keep around one entry per room but that might be after that).